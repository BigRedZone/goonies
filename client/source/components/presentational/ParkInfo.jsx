import React from 'react';
import axios from 'axios';
import Weather from './Weather';


class ParkInfo extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      parkName: '',
      alerts: [],
      info: {},
    };
  }

  componentDidMount() {
    this.getAlerts();
    this.getInfo();
  }

  getAlerts() {
    axios.get('/api/park/alerts')
      .then((response) => {
        this.setState({ alerts: response.data.data }, () => this.render());
      })
      .catch((error) => {
        console.log(error);
      });
  }

  getInfo() {
    axios.get('/api/park/info')
      .then((response) => {
        this.setState({
          info: response.data.data[0],
          parkName: response.data.data[0].name,
        }, () => this.render());
      })
      .catch((error) => {
        console.log(error);
      });
  }

  render() {
    const { parkName } = this.state;
    const { alerts } = this.state;
    const { info } = this.state;
    return (
      <div>
        <h2>
          { parkName }
          <span> </span>
          {info.designation}
        </h2>
        <h3>Alerts</h3>
        {alerts.map((alert) => {
          return (
            <div key={alert.id}>
              <p>
                <strong>{alert.title}</strong>
                <br />
                <span>Category: </span>
                {alert.category}
                <br />
                {alert.description}
              </p>
            </div>
          );
        })}
        <h3>Park Info</h3>
        <div>
          <p>
            {info.description}
            <br />
            <a href={info.url}>Learn More</a>
          </p>
          <p>
            <strong>Directions</strong>
            <br />
            {info.directionsInfo}
            <br />
            <a href={info.directionsUrl}>Driving Directions</a>
          </p>
        </div>
      </div>
    );
  }
}

export default ParkInfo;
