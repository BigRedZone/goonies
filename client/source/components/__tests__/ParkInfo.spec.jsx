import React from 'react';
import renderer from 'react-test-renderer';
import ParkInfo from '../presentational/ParkInfo';

test('App Renders correctly', () => {
  const component = renderer.create(<ParkInfo />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
