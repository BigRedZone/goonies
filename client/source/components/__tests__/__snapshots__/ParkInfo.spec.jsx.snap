// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParkInfo Renders correctly 1`] = `
<div>
  <h2>
    
  </h2>
  <h3>
    Alerts
  </h3>
  <h3>
    Park Info
  </h3>
  <div>
    <p>
      <br />
      <a>
        Learn More
      </a>
    </p>
    <p>
      <strong>
        Directions
      </strong>
      <br />
      <br />
      <a>
        Driving Directions
      </a>
    </p>
  </div>
</div>
`;
